Add store flow to new site segment step

fditrapani opened the pull request:
We recently tested a new site segmenting step that returned really positive results. We've implemented the new screen to 100% of english visitors while we get all the strings translated. With this new step, we no longer have the design-type step and as a result, we don't have any people going through our new store signup flow. 

This PR adds the ability for people to go through the new store signup flow if they pick "Sell products or collect payments", and only "Sell products or collect payments", from under "What’s the primary goal you have for your site?". 

## Screencap
![store-flow](https://user-images.githubusercontent.com/6981253/34059250-326983a0-e1ac-11e7-9cfa-66943892e607.gif)

## Testing store
* Visit `/start`
* **Only** select "Sell products or collect payments" from under "What’s the primary goal you have for your site?". 
* Next you should see a couple themes, pick one and complete signup
* Your should now have a store

## Testing non-store
* Visit `/start`
* Pick "Sell products or collect payments" plus another option from under "What’s the primary goal you have for your site?".
* Next step should be domain, pick a domain and complete signup
* You should now have a regular site

@lamosty @jsnajdr @apeatling @kellychoffman

lamosty commented:
Nice work @fditrapani! Are we okay with dropping the Pressable store flow completely? /cc @apeatling 

Do we have a good event for tracking people who went through the new Store NUX flow? Currently, we rely on `calypso_triforce_select_design` event.

kellychoffman commented:
> Are we okay with dropping the Pressable store flow completely?

Not until we've replaced it, which @allendav is just beginning to work on now.

lamosty commented:
> Not until we've replaced it, which @allendav is just beginning to work on now.

Yeah, I see. As far as I can see, there's no Pressable flow in this PR. It says that if people are not from US/CA, it will just redirect to the current flow name instead of using the new `segmented-store-nux`.

fditrapani commented:
Sure thing @lamosty. I've added the Pressable flow. Unfortunately the only way I could test it was to change the condition here to true: https://github.com/Automattic/wp-calypso/pull/20878/files#diff-7949db3a74b97e79a5ff08e5b794a6d8R314

I also added the tracks event here: https://github.com/Automattic/wp-calypso/pull/20878/files#diff-7949db3a74b97e79a5ff08e5b794a6d8R298

And here's a little screenshot:
![pressable](https://user-images.githubusercontent.com/6981253/34111854-e518faca-e3d8-11e7-8f31-0c352fc19291.gif)

lamosty commented:
> Sure thing @lamosty. I've added the Pressable flow

Cool, thanks for adding it!

> Unfortunately the only way I could test it was to change the condition here to true:

Why do you think so? Make sure you select either the `pressable` or `atomic` variation of `signupAtomicStoreVsPressable` abtest. Other than that, as you are running local Calypso, it doesn't matter if you are from US/CA or not.

allendav commented on client/signup/steps/about/index.jsx:
```diff
@@ -443,57 +489,82 @@ class AboutStep extends Component {
+					<form onSubmit={ this.handleSubmit }>
+						<Card>
+							<FormFieldset>
+								<FormLabel htmlFor="siteTitle">What would you like to name your site?</FormLabel>
```
Please `translate`

fditrapani replied:
Thanks @allendav. This screen is still hidden behind an a/b test which means it will only be shown to English visitors. Since the test passed, I'm in the process of translating all the strings in another PR.

allendav commented on client/signup/steps/about/index.jsx:
```diff
@@ -443,57 +489,82 @@ class AboutStep extends Component {
+								<FormTextInput
+									id="siteTitle"
+									name="siteTitle"
+									placeholder="eg: Mel's Diner, Stevie’s Blog, Vail Renovations"
```
Please `translate`, also `eg` should be `e.g.`

allendav commented on client/signup/steps/about/index.jsx:
```diff
@@ -443,57 +489,82 @@ class AboutStep extends Component {
+							</FormFieldset>
+
+							<FormFieldset>
+								<FormLabel>What will your site be about?</FormLabel>
```
Please `translate`

allendav commented on client/signup/steps/about/index.jsx:
```diff
@@ -443,57 +489,82 @@ class AboutStep extends Component {
+								<FormTextInput
+									id="siteTopic"
+									name="siteTopic"
+									placeholder="eg: Fashion, travel, design, plumber, electrician"
```
Please `translate` and `eg` should be `e.g.`

allendav commented on client/signup/steps/about/index.jsx:
```diff
@@ -443,57 +489,82 @@ class AboutStep extends Component {
+							</FormFieldset>
+
+							<FormFieldset>
+								<FormLabel>What’s the primary goal you have for your site?</FormLabel>
```
Please `translate`

allendav commented on client/signup/utils.js:
```diff
@@ -170,10 +170,14 @@ function getThemeForSiteGoals( siteGoals ) {
 	const siteGoalsValue = siteGoals.split( ',' );
 
-	if ( siteGoalsValue.indexOf( 'sell' ) !== -1 ) {
+	if ( siteGoals === 'sell' ) {
```
Down the road it might be difficult to remember why we handle the singular sell goal differently here. Maybe add a comment before this line explaining the rationale.

fditrapani replied:
Good call, thanks @allendav

fditrapani commented:
Thanks @allendav. I've added the comment as requested but would like to do the translating in another PR. This code/screen is only being shown to English users through an A/B test. Since the test won, I'm working on getting the translations in, in another PR before I remove all the A/B test logic.

fditrapani commented:
> Why do you think so? Make sure you select either the pressable or atomic variation of signupAtomicStoreVsPressable abtest.

Thanks @lamosty this did the job without changing any code :)
